web-app/src/components/jobs/JobDetailModal.tsx

```diff
@@ -0,0 +1,606 @@
+   }
+`;
+const PostedDate = styled.span`
+   color:grey;
+`;
+const ReviewRatingWrapper = styled.div`
+    display:flex;
+    justify-content:space-between;
+    margin-top:10px;
+`;
+const ReviewRating = styled.div`
+    display:flex;
+    flex-direction:column;
+`;
+const ReviewComments = styled.div`
+   display:flex;
+   flex-direction:column;
+`;
+const ReviewCommentsRow = styled(Row)`
+  margin:0px;
+  display:flex;
+  justify-content:space-between;
+`;
+const ReviewCommentsColumn = styled(Col)`
+   border-radius:6px;
+   padding:20px;
+   border:1px solid #efefef;
+   margin:5px;
+   background-color:#fff;
+   position:relative;
+   margin-top:10px;
+   
+`;
+const ReviewCommentHeader = styled(Col)`
+   position:absolute;
+   top: -10px;
+   left: 15px;
+   padding: 2px 10px;
+   background: #efefef;
+   border-radius: 16px;
+`;
+const AveeragePay = styled.div`
+   margin-top:5px;
+   margin-bottom:20px;
+`;
+const GreyText = styled.span`
+   color:grey;
+`;
+const SalaryRangeValues = styled.span`
+   display:flex;
+   justify-content:space-between;
+`;
+const JobLocationList = styled.div`
+   display:flex;
+   flex-direction:column;
+`;
+const JobLocationText = styled.span`
+   
+`;
+const JobList = styled(Row)`
+   margin-top:10px;
+`;
+const RecommendedButton = styled.button`
+   color:${colors.secondaryColorGreen};
+   background-color: #def1de;
+   border-radius: 6px;
+`;
+const ButtonText = styled.div`
+    display:flex;
+    align-items:center;
+`;
+
+Modal.setAppElement('#app-root');
+export default function JobDetailModal(props: JobDetailModalProps): JSX.Element {
+    const { visible, jobPost, onClose } = props
+    const [activetTab, setActiveTab] = useState(1);
+    const jobContentRef = useRef(null);
+    const companyContentRef = useRef(null);
+    return <StyledModal
+        isOpen={visible}
+        onRequestClose={onClose}
+        style={customStyles}
+    >
+        <JobPostContent>
+            <JobPostTitleRow {...jobPost} modal onClose={onClose}></JobPostTitleRow>
+            <Tabs
+                value={activetTab}
+                indicatorColor="primary"
+                textColor="primary"
+                onChange={(e, newValue) => {
+                    if (newValue === 1) {
+                        jobContentRef.current.scrollIntoView({ behavior: 'smooth', block: 'start' })
+                    } else {
+                        companyContentRef.current.scrollIntoView({ behavior: 'smooth', block: 'start' })
+                    }
+                    setActiveTab(newValue);
+                }}
+                aria-label="disabled tabs example"
+            >
+                <Tab value={1} label="Job" />
+                <Tab value={2} label="Company" />
+            </Tabs>
+            <ScrollableContent>
+                <div ref={jobContentRef}>
+                    <JobContentWrapper   {...jobPost}></JobContentWrapper>
+                </div>
+                <CompanyContent ref={companyContentRef} >
```

Comment:
Create a separate component for this similar to how you are using a component for the JobContent